core/src/main/scala/org/apache/spark/sql/execution/CoprocessorRDD.scala

```diff
@@ -374,13 +358,31 @@ case class RegionTaskExec(child: SparkPlan,
                   s", downgrade to table scan with ${tasks.size()} region tasks, " +
                   s"original index scan task has ${taskRanges.size()} ranges, will try to merge."
               )
+              downgrade = true
```

Comment:
It seems that variable `downgrade` has the same semantics as `satisfyDowngradeThreshold`, why do we need to maintain tow of these variables?

Reply:
if checkisunaryrange is true, downgrade will not be proceeded.